plugins/dell-kestrel/fu-dell-kestrel-ec.c

```diff
@@ -0,0 +1,1079 @@
+/*
+ * Copyright 2024 Dell Technologies
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-dell-kestrel-common.h"
+
+typedef struct __attribute__((packed)) { /* nocheck:blocked */
+	guint8 dock_configuration;
+	guint8 dock_type;
+	guint16 power_supply_wattage;
+	guint16 module_type;
+	guint16 board_id;
+	guint16 port0_dock_status;
+	guint16 port1_dock_status; // unused for K2 dock, should be 0
+	guint32 dock_firmware_pkg_ver;
+	guint64 module_serial;
+	guint64 original_module_serial;
+	guint8 service_tag[7];
+	guint8 marketing_name[32];
+	guint32 dock_error;
+	guint32 dock_module_status;
+	guint32 dock_module_error;
+	guint8 reserved;
+	guint32 dock_status;
+	guint16 dock_state;
+	guint16 dock_config;
+	guint8 dock_mac_addr[6];
+	guint32 dock_capabilities;
+	guint32 dock_policy;
+	guint32 dock_temperature;
+	guint32 dock_fan_speed;
+	guint16 upf_power;
+	guint8 eppid;
+	guint8 unused[74];
+} FuDellKestrelDockDataStructure;
+
+typedef struct __attribute__((packed)) { /* nocheck:blocked */
+	struct FuDellKestrelDockInfoHeader {
+		guint8 total_devices;
+		guint8 first_index;
+		guint8 last_index;
+	} header;
+	struct FuDellKestrelEcQueryEntry {
+		struct FuDellKestrelEcAddrMap {
+			guint8 location;
+			guint8 device_type;
+			guint8 sub_type;
+			guint8 arg;
+			guint8 instance;
+		} ec_addr_map;
+		union {
+			guint32 version_32;
+			guint8 version_8[4];
+		} __attribute__((packed)) version; /* nocheck:blocked */
```

Comment:
I guess the backport for the rs stuff was tough?  I'm fine with this knowing we have it done right in main, but I think @hughsie can make the call here.

Reply:
> I guess the backport for the rs stuff was tough? 

I tried to pull the rsgen from main but several differencies are seen in the history, likely the backport are selected to apply. 

For broader stability on a stable branch I keep the changes in plugin alone as is.  See @hughsie if he plan to backport the new work in main.

Reply:
I had a quick glance and it looks like a bunch of gymnastics to make it happen, maybe better for you to.

Reply:
How about let's merge this PR and tag 1.9.27 to address the schedule and stability concerns?  Another tag with sufficient time for the rs work and adoption/testing.

Reply:
If we do that the follow-on work never gets done.